crate/kmip/src/kmip/kmip_operations.rs

```diff
@@ -1237,6 +1242,43 @@ pub struct RevokeResponse {
     pub unique_identifier: UniqueIdentifier,
 }
 
+/// This request is used to generate a replacement key for an existing symmetric key. It is analogous to the Create operation, except that attributes of the replacement key are copied from the existing key, with the exception of the attributes listed in Re-key Attribute Requirements.
+///
+/// As the replacement key takes over the name attribute of the existing key, Re-key SHOULD only be performed once on a given key.
+///
+/// The server SHALL copy the Unique Identifier of the replacement key returned by this operation into the ID Placeholder variable.
+///
+/// For the existing key, the server SHALL create a Link attribute of Link Type Replacement Object pointing to the replacement key. For the replacement key, the server SHALL create a Link attribute of Link Type Replaced Key pointing to the existing key.
+///
+/// An Offset MAY be used to indicate the difference between the Initial Date and the Activation Date of the replacement key. If no Offset is specified, the Activation Date, Process Start Date, Protect Stop Date and Deactivation Date values are copied from the existing key.
+#[derive(Debug, Serialize, Deserialize, Default, PartialEq, Eq)]
+#[serde(rename_all = "PascalCase")]
+pub struct ReKey {
+    // Determines the existing Symmetric Key being re-keyed. If omitted, then the ID Placeholder value is used by the server as the Unique Identifier.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub unique_identifier: Option<UniqueIdentifier>,
```

Comment:
Remove option here?

Reply:
I'd rather let it be optional be conform the KMIP specs